### ISSUE TITLE
obj: pmemobj_list_insert does not return any errno when offset is extremely big

### ISSUE THREAD

**Issue by mglaner** [pmem/issues#772]
pmemobj_list _insert
pmemobj_list_insert_new
pmemobj_list_move
pmemobj_list_remove
Following API functions do not return any errno when passing extremely big offset (maximum range of type unsigned long long) and valid other arguments.

Described situation occures untill this change in the library:
SHA-1: 4ee9be7dce75d9cb93db6fde363b0060b227a443 (obj: implement automatic pool extending)

Found on 1.3+b2-467-g4ee9be7

**Comment by pbalcer** [comment]
There's no way to check for valid offset. We can only check if it's one of the extremes that are known to be invalid. So if you call this function with `offset = rand() % POOL_SIZE;` that check won't do a thing.
IMHO such assertions are pointless and serve only to satisfy overzealous tests.

**Comment by krzycz** [comment]
@pbalcer - Well, I think it's pretty easy to catch offsets that are invalid. Of course, it may happen that some random value, by accident, falls into the valid range, but the same may happen with any other argument of any function.
BTW, as I understand, the valid offset is not the offset within <0; pool_size> range, but within <0; object_size - sizeof(struct list_elem)>. So, we only need to do is to determine the target object size - it's the cost of pmemobj_alloc_usable_size(), isn't it?

**Comment by pbalcer** [comment]
pe_offset is the list_entry offset relative to the target object. The target object is technically speaking unknown at the time of the call.

**Comment by krzycz** [comment]
Perhaps I'm missing something, but it seems for me that the target object is known - it's pointed by *oid*, so we can find its size. It's even easier in case of *pmemobj_list_insert_new*, as the size is the argument of this function.

**Comment by krzycz** [comment]
See the proposed solution under the link above.

**Comment by pbalcer** [comment]
Mea culpa, you are right. I was thinking about head/tail objects on the list having to be calculated and only after that we could have called usable_size().

**Comment by mglaner** [comment]
Verified on 1.3+b2-815-g91158da
